src/server/middleware/systraceMiddleware.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright 2017-present, Callstack.
+ * All rights reserved.
+ * 
+ * @flow
+ */
+import type { Middleware, $Request } from 'express';
+
+const fs = require('fs');
+const dedent = require('dedent');
+const logger = require('../../logger');
+
+const systraceMiddleware: Middleware = (req: $Request, res, next) => {
+  if (req.path !== '/systrace') {
+    next();
+    return;
+  }
+
+  const path = `/tmp/react_native_${Date.now()}.json`;
+
+  fs.writeFile(path, req.rawBody, err => {
+    if (err) {
+      next(err);
+      return;
+    }
+
+    logger.debug('Systrace', `/tmp/react_native_${Date.now()}.json`);
+
+    res.end(
+      dedent`
+      Your trace report was saved at ${path}.
+      
+      You can open it with Google Chrome by navigating to 'chrome://tracing'
+      and clicking 'load'.
+
+      The path to the report was printed by Haul in your Terminal so you can
```

Comment:
I think that given our logging system that highlights type of log, it makes sense to stay in the first line.

<img width="646" alt="screen shot 2017-04-05 at 15 34 03" src="https://cloud.githubusercontent.com/assets/2464966/24707845/66d1ec2c-1a15-11e7-8680-d97d06436a8d.png">